x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/ConfigInitialNode.java

```diff
@@ -82,83 +88,81 @@
+    private static final String TLS_CONFIG_DIR_NAME_PREFIX = "tls_auto_config_initial_node_";
 
     public ConfigInitialNode() {
         super("Generates all the necessary security configuration for the initial node of a new secure cluster");
+        // This "cli utility" must be invoked EXCLUSIVELY from the node startup script, where it is passed all the
+        // node startup options unfiltered. It cannot consume most of them, but it does need to inspect the `-E` ones.
+        parser.allowsUnrecognizedOptions();
     }
 
     public static void main(String[] args) throws Exception {
         exit(new ConfigInitialNode().main(args, Terminal.DEFAULT));
     }
 
+    private boolean isInitialNodeAutoConfigured(Environment environment) {
+        return XPackSettings.SECURITY_ENABLED.get(environment.settings()) &&
+                XPackSettings.ENROLLMENT_ENABLED.get(environment.settings()) &&
+                isInitialClusterNode(environment.settings()) &&
+                DiscoveryNode.isMasterNode(environment.settings()) &&
+                false == DiscoveryNode.hasRole(environment.settings(), DiscoveryNodeRole.VOTING_ONLY_NODE_ROLE) &&
+                DiscoveryNode.canContainData(environment.settings()) &&
+                usesTransportTLSWithAutoGeneratedCerts(environment);
+    }
+
     @Override
     protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
         // Silently skipping security auto configuration because node considered as restarting.
         if (Files.isDirectory(env.dataFile()) && Files.list(env.dataFile()).findAny().isPresent()) {
-            terminal.println(expectedNoopVerbosityLevel(),
+            terminal.println(Terminal.Verbosity.VERBOSE,
                     "Skipping security auto configuration because it appears that the node is not starting up for the first time.");
-            terminal.println(expectedNoopVerbosityLevel(),
+            terminal.println(Terminal.Verbosity.VERBOSE,
                     "The node might already be part of a cluster and this auto setup utility is designed to configure Security for new " +
                             "clusters only.");
-            if (options.has(strictOption)) {
-                throw new UserException(ExitCodes.NOOP, null);
-            } else {
-                return; // silent error because we wish the node to start as usual (skip auto config) during a restart
-            }
+            // we wish the node to start as usual during a restart
+            // but still the exit code should indicate that this has not been run
+            throw new UserException(ExitCodes.NOOP, null);
+        }
+        // the node looks like it is already configured in an automatic fashion, but the node did not start
+        // at the point when this was started
+        // return exit code 0 here to signal that credential generation should be attempted
+        if (isInitialNodeAutoConfigured(env)) {
```

Comment:
Yes, you are right about the flow.
I kept this code around in order to always exit with 0 if node is auto configured, and non-zero otherwise. Since nobody cares about the exist codes anymore (unless it is fatal) then if we want to minimize code this can be removed.

Your call.

Reply:
> Your call.

Shouldn't be. It's a suggestion, I have an opinion that this code is redundant now. If you think that

> I kept this code around in order to always exit with 0 if node is auto configured, and non-zero otherwise

has value, let's keep this as is.

Reply:
I removed this code